node/src/ActiveSpeakerObserver.ts

```diff
@@ -28,12 +28,12 @@ export interface ActiveSpeakerObserverActivity
 
 export type ActiveSpeakerObserverEvents = RtpObserverEvents &
 {
-	dominantspeaker: [{ producer: Producer }];
+	dominantspeaker: [Producer];
```

Comment:
This is still an array despite changelog claims, what is the point of these changes?

Reply:
It's not an array, it's the syntax of the arguments of the emitted events. Example:

<img width="592" alt="CleanShot-2022-11-05-at-01 13 08@2x" src="https://user-images.githubusercontent.com/16191/200091739-ce42c8c8-bfd9-4561-b6e3-c779ae74d03a.png">

Reply:
I see, so the comment about array is misleading.

Well, I'd argue that it was better to fix the documentation rather than breaking API for everyone. People relied on `.producer` field of an object, now their code will stop working.

Reply:
For TypeScript users (in strict mode) this will stop compiling, for JS users this will potentially break their apps.

Reply:
The Rust code mimics the corresponding TS code, but the TS code didn't match the expectation (API docs).

Reply:
Things reverted in https://github.com/versatica/mediasoup/pull/947